cosmic-settings/src/pages/networking/wifi.rs

```diff
@@ -0,0 +1,668 @@
+                .push(widget::list_column().add(wifi_enable))
+                .push_maybe(state.airplane_mode.then(|| {
+                    widget::container(widget::text::body(&section.descriptions[airplane_mode_txt]))
+                        .align_x(alignment::Horizontal::Center)
+                }));
+
+            if !state.airplane_mode
+                && state.known_access_points.is_empty()
+                && state.wireless_access_points.is_empty()
+            {
+                let no_networks_found =
+                    widget::container(widget::text::body(&section.descriptions[no_networks_txt]))
+                        .align_x(alignment::Horizontal::Center);
+
+                view = view.push(no_networks_found);
+            } else {
+                let mut has_known = false;
+                let mut has_visible = false;
+
+                // Create separate sections for known and visible networks.
+                let (known_networks, visible_networks) = state.wireless_access_points.iter().fold(
+                    (
+                        widget::settings::view_section(&section.descriptions[known_networks_txt]),
+                        widget::settings::view_section(&section.descriptions[visible_networks_txt]),
+                    ),
+                    |(mut known_networks, mut visible_networks), network| {
+                        let is_connected = is_connected(state, network);
+                        let (connect_txt, connect_msg) = if is_connected {
+                            (&section.descriptions[connected_txt], None)
+                        } else if page.connecting.contains(&network.ssid) {
+                            (&section.descriptions[connecting_txt], None)
+                        } else {
+                            (
+                                &section.descriptions[connect_txt],
+                                Some(Message::Connect(network.ssid.clone())),
+                            )
+                        };
+
+                        let is_known = state
+                            .known_access_points
+                            .iter()
+                            .map(|known| known.ssid.as_ref())
+                            .chain(state.active_conns.iter().filter_map(|active| {
+                                if let ActiveConnectionInfo::WiFi { name, .. } = active {
+                                    Some(name.as_str())
+                                } else {
+                                    None
+                                }
+                            }))
+                            .any(|known| known == network.ssid.as_ref());
+
+                        // TODO: detect if access point is secured or not.
+                        let is_encrypted = true;
+
+                        let identifier = widget::row::with_capacity(3)
+                            .push(widget::icon::from_name("network-wireless-good-symbolic"))
```

Comment:
This should possibly be `network-wireless-signal-good-symbolic` (maybe excellent instead of good). Though the icons should probably change depending on signal strength, similar to the network applet.

Reply:
[This](https://www.figma.com/design/yFxlLCtVYLHMuHBNSXLf7O/Icon-theme?node-id=409-3568&t=afKFZMv5n2XfqkCg-4) icon needs to be added to the icon theme as well (and used in the **Add network** button), but I don't want to spam PRs.